nisystemlink/clients/testmonitor/utilities/_dataframe_utilities.py

```diff
@@ -183,59 +188,51 @@ def __is_property_header(header: str) -> bool:
-            data parameter has the required fields.
+        is_valid_measurement: Optional callback function that checks if a step data
+            parameter is a valid measurement so that only those are included in the returned dataframe.
 
     Returns:
         List[Dict[str, Any]]: A list of dictionaries containing step information.
     """
     steps_dict = []
     for step in steps:
-        single_step_dict = step.dict(exclude_none=True)
+        __filter_invalid_measurements(step, is_valid_measurement)
 
-        __filter_data_parameters(single_step_dict, step, is_measurement_data_parameter)
+        single_step_dict = step.dict(exclude_none=True)
         __normalize_inputs_outputs(single_step_dict, step)
         __normalize_step_status(single_step_dict)
         steps_dict.append(single_step_dict)
     return steps_dict
 
 
-def __filter_data_parameters(
-    step_dict: Dict[str, Any],
+def __filter_invalid_measurements(
     step: Step,
-    is_measurement_data_parameter: Optional[Callable[[Dict[str, Any]], bool]],
+    is_valid_measurement: Optional[Callable[[Measurement], bool]],
 ) -> None:
     """Gets data parameters from the step dictionary and filters it based on the callback function.
 
     Args:
-        step_dict: A dictionary with step information.
         step: A Step object containing data parameters.
-        is_measurement_data_parameter: Optional callback function to check if a measurement is valid. The method takes
+        is_valid_measurement: Optional callback function to check if a measurement is valid. The method takes
             a dictionary as input and returns a boolean value. The default behavior is to consider only parameters
-            that have both 'name' and 'measurement' fields as measurement parameters.
+            that have both 'name' and 'measurement' fields with values as valid measurements.
 
     Returns:
-        None: The function modifies step_dict in place with the filtered step data parameters.
+        None: The function modifies step parameters in place with filtered data parameters.
     """
-    valid_measurement_parameters = []
-    if (
-        is_measurement_data_parameter is not None
-        and step.data is not None
-        and step.data.parameters is not None
-    ):
-        for parameter in step_dict["data"]["parameters"]:
-            if is_measurement_data_parameter and is_measurement_data_parameter(
-                parameter
-            ):
-                valid_measurement_parameters.append(parameter)
+    if step.data and step.data.parameters and is_valid_measurement is not None:
+        valid_measurement_parameters = []
+        for measurement in step.data.parameters:
+            if is_valid_measurement and is_valid_measurement(measurement):
+                valid_measurement_parameters.append(measurement)
 
-        step_dict["data"]["parameters"] = valid_measurement_parameters
+        step.data.parameters = valid_measurement_parameters
```

Comment:
This is going to modify the Step that was input to the conversion, and we don't want that. Those inputs should be left unchanged.